### ISSUE TITLE
Seems to be sometimes linting files in .git

### ISSUE THREAD

**Issue by kentor** [rizowski/eslint-watch#52]
Using Node 4.2.1.

I have a directory with

index.js
```js
({
    a: 'arg1',
    b: 'arg2'
})
```

package.json
```json
{
  "devDependencies": {
    "eslint": "^1.9.0",
    "eslint-watch": "^2.1.3"
  }
}
```

.eslintrc
```yaml
---
rules:
  comma-dangle: [2, always]
```

After initiating this directory with git and committing the first commit, if I start watching and I run these commands in another terminal:

```bash
❯ echo "adsf" >> index.js

eslint-test git/master*
❯ echo "adsf" >> index.js

eslint-test git/master*
❯ git checkout -- .
```

I see

```
❯ $(npm bin)/esw -w
/Users/kenneth/stuff/eslint-test/index.js (1/0)
  ✖  3:14  Missing trailing comma  comma-dangle

✖ 1 error (10:12:38 AM)


/Users/kenneth/stuff/eslint-test/index.js (1/0)
  ✖  3:14  Missing trailing comma  comma-dangle

✖ 1 error (10:12:49 AM)


/Users/kenneth/stuff/eslint-test/index.js (1/0)
  ✖  3:14  Missing trailing comma  comma-dangle

✖ 1 error (10:12:55 AM)


/Users/kenneth/stuff/eslint-test/.git/index (1/0)
  ✖  1:6  Parsing error: Unexpected token ILLEGAL

✖ 1 error (10:12:55 AM)
```

**Comment by rizowski** [comment]
Strange... I will take a look at this. There should be a rule checking to see if the path is ignored before it tries to lint it, but maybe that is failing somehow.

**Comment by rizowski** [comment]
@kentor What OS are you using for this to appear?

**Comment by kentor** [comment]
osx 10.10.3

here's some logs from `DEBUG=* $(npm bin)/esw -w 2>&1 | tee log`

```
Thu, 12 Nov 2015 05:11:14 GMT esw:eslint-cli Loaded
Thu, 12 Nov 2015 05:11:14 GMT esw:eslint-cli EsLint path: /Users/kenneth/stuff/eslint-test/node_modules/.bin/eslint
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Loaded
Thu, 12 Nov 2015 05:11:15 GMT esw:options Loaded
Thu, 12 Nov 2015 05:11:15 GMT esw:watcher Loaded
Thu, 12 Nov 2015 05:11:15 GMT esw:arg-parser Loaded
Thu, 12 Nov 2015 05:11:15 GMT esw:esw-cli Loaded
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Executing help
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-cli Linting: undefined
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Help text received
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Alias found: -c
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --config
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --no-eslintrc
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --env
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --ext
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --global
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --parser
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --cache
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --cache-file
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --cache-location
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --rulesdir
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --plugin
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --rule
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --ignore-path
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --no-ignore
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --ignore-pattern
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --stdin
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --stdin-filename
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --quiet
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --max-warnings
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Alias found: -o
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --output-file
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Alias found: -f
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --format
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --no-color
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --init
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --fix
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --debug
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Alias found: -h
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --help
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Alias found: -v
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-help Parsing --version
Thu, 12 Nov 2015 05:11:15 GMT esw:esw-cli Arguments passed: [ '/Users/kenneth/.nvm/versions/node/v4.2.1/bin/node', '/Users/kenneth/stuff/eslint-test/node_modules/.bin/esw', '-w' ]
Thu, 12 Nov 2015 05:11:15 GMT esw:esw-cli Parsing args
Thu, 12 Nov 2015 05:11:15 GMT esw:arg-parser Directories to check: []
Thu, 12 Nov 2015 05:11:15 GMT esw:arg-parser Contains: /Users/kenneth/.nvm/versions/node/v4.2.1/bin/node
Thu, 12 Nov 2015 05:11:15 GMT esw:arg-parser Contains: /Users/kenneth/stuff/eslint-test/node_modules/.bin/esw
Thu, 12 Nov 2015 05:11:15 GMT esw:arg-parser setting custom formatter
Thu, 12 Nov 2015 05:11:15 GMT esw:arg-parser GetPath: simple-detail
Thu, 12 Nov 2015 05:11:15 GMT esw:arg-parser Setting default path: ./
Thu, 12 Nov 2015 05:11:15 GMT esw:esw-cli Running initial lint
Thu, 12 Nov 2015 05:11:15 GMT esw:esw-cli [ '-f', '/Users/kenneth/stuff/eslint-test/node_modules/eslint-watch/src/formatters/simple-detail', './' ]
Thu, 12 Nov 2015 05:11:15 GMT esw:eslint-cli Linting: './'
Thu, 12 Nov 2015 05:11:15 GMT esw:esw-cli -w seen
Thu, 12 Nov 2015 05:11:15 GMT esw:watcher Watching: './'
Thu, 12 Nov 2015 05:11:16 GMT eslint:cli Running on files
Thu, 12 Nov 2015 05:11:16 GMT eslint:glob-util Creating list of files to process.
Thu, 12 Nov 2015 05:11:16 GMT eslint:cli-engine Processing /Users/kenneth/stuff/eslint-test/index.js
Thu, 12 Nov 2015 05:11:16 GMT eslint:cli-engine Linting /Users/kenneth/stuff/eslint-test/index.js
Thu, 12 Nov 2015 05:11:16 GMT eslint:config Constructing config for /Users/kenneth/stuff/eslint-test/index.js
Thu, 12 Nov 2015 05:11:16 GMT eslint:config Using .eslintrc and package.json files
Thu, 12 Nov 2015 05:11:16 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:16 GMT eslint:config Using /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:16 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/package.json
Thu, 12 Nov 2015 05:11:16 GMT eslint:config Merging command line environment settings
Thu, 12 Nov 2015 05:11:16 GMT eslint:cli-engine Linting complete in: 50ms
/Users/kenneth/stuff/eslint-test/index.js (1/0)
  ✖  3:14  Missing trailing comma  comma-dangle

✖ 1 error (9:11:16 PM)

Thu, 12 Nov 2015 05:11:16 GMT esw:esw-cli Setting exit code to: 1
Thu, 12 Nov 2015 05:11:24 GMT esw:watcher Changed: index.js
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Constructing config for index.js
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Using .eslintrc and package.json files
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Using /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/package.json
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Merging command line environment settings
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Merging command line rules
Thu, 12 Nov 2015 05:11:24 GMT esw:watcher lintFile: index.js
Thu, 12 Nov 2015 05:11:24 GMT eslint:glob-util Creating list of files to process.
Thu, 12 Nov 2015 05:11:24 GMT eslint:cli-engine Processing /Users/kenneth/stuff/eslint-test/index.js
Thu, 12 Nov 2015 05:11:24 GMT eslint:cli-engine Linting /Users/kenneth/stuff/eslint-test/index.js
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Constructing config for /Users/kenneth/stuff/eslint-test/index.js
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Using .eslintrc and package.json files
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Using /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/package.json
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Merging command line environment settings
Thu, 12 Nov 2015 05:11:24 GMT eslint:config Merging command line rules
Thu, 12 Nov 2015 05:11:24 GMT eslint:cli-engine Linting complete in: 36ms
Thu, 12 Nov 2015 05:11:24 GMT esw:watcher result: { filePath: '/Users/kenneth/stuff/eslint-test/index.js', messages: [ { ruleId: 'comma-dangle', severity: 2, message: 'Missing trailing comma.', line: 3, column: 14, nodeType: 'Property', source: '    b: \'arg2\'' } ], errorCount: 1, warningCount: 0 }

/Users/kenneth/stuff/eslint-test/index.js (1/0)
  ✖  3:14  Missing trailing comma  comma-dangle

✖ 1 error (9:11:24 PM)

Thu, 12 Nov 2015 05:11:28 GMT esw:watcher Changed: index.js
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Constructing config for index.js
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Using .eslintrc and package.json files
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Using /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/package.json
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Merging command line environment settings
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Merging command line rules
Thu, 12 Nov 2015 05:11:28 GMT esw:watcher lintFile: index.js
Thu, 12 Nov 2015 05:11:28 GMT eslint:glob-util Creating list of files to process.
Thu, 12 Nov 2015 05:11:28 GMT eslint:cli-engine Processing /Users/kenneth/stuff/eslint-test/index.js
Thu, 12 Nov 2015 05:11:28 GMT eslint:cli-engine Linting /Users/kenneth/stuff/eslint-test/index.js
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Constructing config for /Users/kenneth/stuff/eslint-test/index.js
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Using .eslintrc and package.json files
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Using /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/package.json
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Merging command line environment settings
Thu, 12 Nov 2015 05:11:28 GMT eslint:config Merging command line rules
Thu, 12 Nov 2015 05:11:28 GMT eslint:cli-engine Linting complete in: 6ms
Thu, 12 Nov 2015 05:11:28 GMT esw:watcher result: { filePath: '/Users/kenneth/stuff/eslint-test/index.js', messages: [ { ruleId: 'comma-dangle', severity: 2, message: 'Missing trailing comma.', line: 3, column: 14, nodeType: 'Property', source: '    b: \'arg2\'' } ], errorCount: 1, warningCount: 0 }

/Users/kenneth/stuff/eslint-test/index.js (1/0)
  ✖  3:14  Missing trailing comma  comma-dangle

✖ 1 error (9:11:28 PM)

Thu, 12 Nov 2015 05:11:35 GMT esw:watcher Changed: .git/index
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Constructing config for .git/index
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Using .eslintrc and package.json files
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Using /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/package.json
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Merging command line environment settings
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Merging command line rules
Thu, 12 Nov 2015 05:11:35 GMT esw:watcher lintFile: .git/index
Thu, 12 Nov 2015 05:11:35 GMT eslint:glob-util Creating list of files to process.
Thu, 12 Nov 2015 05:11:35 GMT eslint:cli-engine Processing /Users/kenneth/stuff/eslint-test/.git/index
Thu, 12 Nov 2015 05:11:35 GMT eslint:cli-engine Linting /Users/kenneth/stuff/eslint-test/.git/index
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Constructing config for /Users/kenneth/stuff/eslint-test/.git/index
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Using .eslintrc and package.json files
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Using /Users/kenneth/stuff/eslint-test/.eslintrc
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Loading /Users/kenneth/stuff/eslint-test/package.json
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Merging command line environment settings
Thu, 12 Nov 2015 05:11:35 GMT eslint:config Merging command line rules
Thu, 12 Nov 2015 05:11:35 GMT eslint:cli-engine Linting complete in: 3ms
Thu, 12 Nov 2015 05:11:35 GMT esw:watcher result: { filePath: '/Users/kenneth/stuff/eslint-test/.git/index', messages: [ { fatal: true, severity: 2, message: 'Parsing error: Unexpected token ILLEGAL', line: 1, column: 6 } ], errorCount: 1, warningCount: 0 }

/Users/kenneth/stuff/eslint-test/.git/index (1/0)
  ✖  1:6  Parsing error: Unexpected token ILLEGAL

✖ 1 error (9:11:35 PM)

Thu, 12 Nov 2015 05:11:40 GMT esw:esw-cli c was pressed
```

**Comment by rizowski** [comment]
@kentor I pushed up a branch with `.git, node_modules, bower_components` being in the ignore paths. Will you let me know if that fixes your problem? If it does I will push it out as a release.

**Comment by kentor** [comment]
I think I've been getting it for other directories besides those three, but I would have to do some more extensive testing. I rather we have a proper fix than a bandaid, so I'll do some more testing when I have time.

**Comment by rizowski** [comment]
@kentor I just released 2.1.4. Let me know if you see any other problems.

**Comment by kentor** [comment]
thanks!
